fix 'canister' issue

acattoXD opened the pull request:
This pull request fixes "canister" being broken on GIR.

acattoXD committed:
fix 'canister' issue

HydrationMan reviewed (approved):
Doesn't _look_ like malware anyway :trollface:

SlimShadyIAm commented on utils/views/menus/canister.py:
```diff
@@ -170,49 +170,71 @@ async def canister(ctx: GIRContext, whisper: bool, result):
 class TweakDropdown(discord.ui.Select):
     def __init__(self, author, entries, interaction, should_whisper):
         self.repo_icons = {
```
Should probably rename this vaiable to make it clear what this does now

SlimShadyIAm commented on utils/views/menus/canister.py:
```diff
@@ -170,49 +170,71 @@ async def canister(ctx: GIRContext, whisper: bool, result):
+        for index, option in enumerate(entries):
             if option.get("package") in seen_packages:
                 continue
+                
```
What is the purpose of this change? Why not just use the UUID?

SlimShadyIAm replied:
This comment is about L199

SlimShadyIAm reviewed (changes requested):
Can you explain what you did and the purpose of the changes?

corruptedkiwi commented on utils/views/menus/canister.py:
```diff
@@ -170,49 +170,71 @@ async def canister(ctx: GIRContext, whisper: bool, result):
+            "https://repo.theodyssey.dev": {"icon": "<:colored_sileo_tweak_icon_odyssey:957471133844799518>", "priority": 7},
+            "https://repo.packix.com": {"icon": "<:colored_sileo_tweak_icon_packix:957471186638479470>", "priority": 8},
+            "https://apt.procurs.us": {"icon": "<:colored_sileo_tweak_icon_procurs:957471236718460959>", "priority": 9},
+            "https://repo.twickd.com": {"icon": "<:colored_sileo_tweak_icon_twickd:957471321351135233>", "priority": 10},
```
Since you're creating a set order, you should put everything into an order that makes sense. Here's what I would do:

```
self.repo_icons = {
            "https://havoc.app": {"icon": "<:colored_sileo_tweak_icon_havoc:957471075942428723>", "priority": 1},
            "https://repo.chariz.com": {"icon": "<:colored_sileo_tweak_icon_chariz:957470675411537980>", "priority": 2},
            "http://apt.thebigboss.org/repofiles/cydia": {"icon": "<:colored_sileo_tweak_icon_bigboss:957470630956138496>", "priority": 3},
            "https://apt.procurs.us": {"icon": "<:colored_sileo_tweak_icon_procurs:957471236718460959>", "priority": 4},
            "https://repo.theodyssey.dev": {"icon": "<:colored_sileo_tweak_icon_odyssey:957471133844799518>", "priority": 5},
            "https://repo.chimera.sh": {"icon": "<:colored_sileo_tweak_icon_chimera:957470798602440714>", "priority": 6},
            "https://apt.bingner.com": {"icon": "<:colored_sileo_tweak_icon_elu:957471017578684496>", "priority": 7},
            "https://repo.twickd.com": {"icon": "<:colored_sileo_tweak_icon_twickd:957471321351135233>", "priority": 8},
            "https://repo.packix.com": {"icon": "<:colored_sileo_tweak_icon_packix:957471186638479470>", "priority": 9},
            "https://repo.dynastic.co": {"icon": "<:colored_sileo_tweak_icon_dynasti:957470833952043038>", "priority": 10},
        }
```

I'm not sure why these ten repos were explicitly chosen (or, rather, why it is still these ten), but this is my order. I put Havoc, Chariz, and BigBoss first, with Havoc and Chariz being above BigBoss all for obvious reasons. Then, the four bootstrap repos, with Procursus having the most priority. Then Dynastic, Packix, and Twickd, as they are all deprecated, are at the bottom of the list ordered by the date of the last tweak added, but the specific order isn't as important.

acattoXD committed:
changed priority + option.get("id")

changed priority (thanks kiwi), changed it from a randomly generated uuid (yes, that worked) to whatever canister returned by using option.get("id")

acattoXD committed:
Changed function name of repo_icons

Changed the function name of repo_icons to repo_icons_and_priority

bradleytechman reviewed (approved):
it looks fine and not malware

dotechod reviewed (approved):
looks like a solution to me

bradleytechman commented:
Please merge 🙏🏻

SlimShadyIAm commented:
![image](https://github.com/user-attachments/assets/afa59e86-e7d8-44da-bfee-a8ae9edb7571)
